gamemodes/terrortown/gamemode/server/sv_roleselection.lua

```diff
@@ -782,30 +778,21 @@ end
 local function SelectBaseRolePlayers(plys, subrole, roleAmount)
 	local curRoles = 0
 	local plysList = {}
+	local roleData = roles.GetByIndex(subrole)
 
-	local minKarmaCVar = GetConVar("ttt_" .. roles.GetByIndex(subrole).name .. "_karma_min")
-	local min_karmas = minKarmaCVar and minKarmaCVar:GetInt() or 0
-
-	while curRoles < roleAmount and #plys > 0 do
```

Comment:
Doesnt removing this while loop here means, that players might not get a role in this pass?

As far as I can see, the "CanSelectRole" method can most often just return false.

Reply:
(After requested Review)
Isnt this still having the same problem here?
Exchanging the while loop with a for loop seems like different behaviour?